dojo/tools/blackduck/importer.py

```diff
@@ -0,0 +1,78 @@
+from typing import Iterable, Mapping, Set, Tuple
+from dojo.tools.blackduck.model import BlackduckFinding
+import csv
+import io
+import zipfile
+
+
+class Importer(ABC):
+    @abstractmethod
+    def parse_findings(self, report: Path) -> Iterable[BlackduckFinding]:
+        while False:
+            yield None
+
+
+class BlackduckImporter(Importer):
+    def parse_findings(self, report: Path) -> Iterable[BlackduckFinding]:
+        files = dict()
+        security_issues = dict()
+        try:
+            if zipfile.is_zipfile(report):
+                print("This is a zip file")
+                with zipfile.ZipFile(report) as zip:
+                    for file_name in zip.namelist():
+                        if file_name.endswith('files.csv'):
+                            print("dealing with files")
+                            with io.TextIOWrapper(zip.open(file_name), newline='') as f:
+                                files = self.__partition_by_project_id(f)
+                        elif file_name.endswith('security.csv'):
+                            print("dealing with security")
+                            with io.TextIOWrapper(zip.open(file_name), newline='') as f:
+                                security_issues = self.__partition_by_project_id(f)
+            else:
+                print("Not a zip file?")
+                print(report)
+                # raise ValueError
+        except Exception as e:
+            print("Could not process zip file: {}".format(e))
+
+        project_ids = set(files.keys()) & set(security_issues.keys())
+        for project_id in project_ids:
+            locations = set()
+            for file_entry in files[project_id]:
+                path = file_entry[8]
+                archive_context = file_entry[9]
+                if archive_context:
+                    locations.add(f"{archive_context}{path[1:]}")
+                else:
+                    locations.add(path)
+            for issue in security_issues[project_id]:
+                yield BlackduckFinding(
+                    issue[8],  # vuln ID
+                    issue[9],  # description
+                    issue[21],  # security_risk
+                    issue[14],  # impact
+                    issue[15],  # vulnerability source
+                    issue[20],  # url
+                    issue[6],  # channel version origin id
+                    issue[10],  # published date
+                    issue[11],  # updated on
+                    issue[12],  # base score
+                    issue[13],  # exploitability
+                    issue[16],  # remediation status
+                    issue[17],  # remediation target date
+                    issue[18],  # remediation actual date
+                    issue[19],  # remediation comment
+                    ', '.join([location for location in locations])
+                )
+
+    def __partition_by_project_id(self, csv_file) -> Mapping[str, Set[Tuple[str]]]:
```

Comment:
The return type of this function might not work correctly in 3.5. See https://docs.python.org/3/whatsnew/3.6.html#typing

If that's the case, you can just change the line to this:

```suggestion
    # returns Mapping[str, Set[Tuple[str]]]
    def __partition_by_project_id(self, csv_file):
```

I'll also note that I've since discovered the `Union` type which would allow for fixing the `csv_file` parameter type to be something like `Union[Path, zipfile.ZipFile]`, though I haven't tested that out. Given that this is a "private" method, the types aren't really part of the public API anyways, so it's not that big a deal.